### DISCUSSION TITLE
Specific Database grant role

### DISCUSSION THREAD

**Discussion by scoooobert** [milvus-io/milvus#31923] [Q&A and General discussion]
I want to create a role that allows me to perform all actions within the database but prevents me from modifying any user or role settings. After creating the role, I encountered an issue where I couldn't access the database when logged in using Attu. To resolve this, I granted 'global *' privileges, allowing me to log in to the database. How should I configure this properly?

**Comment by yhmo** [comment]
Read this pdf to know how to crate a role for a specific database:
[roles_users.pdf](https://github.com/milvus-io/milvus/files/14895561/roles_users.pdf)

**Reply by scoooobert** [reply]
I set up following your instructions, but when I was trying to log in, it showed:
![Screenshot 2024-04-07 130557](https://github.com/milvus-io/milvus/assets/92564809/87a95256-9048-4e7d-b758-d859b41709d2)
These are the configurations I did:
![Screenshot 2024-04-07 130727](https://github.com/milvus-io/milvus/assets/92564809/9d869c19-7f1c-4aaf-a2c6-0b838aee6d25)

**Reply by yhmo** [reply]
Login with the root user, edit the role_b/role_b, and add a privilege "ListDatabase" to them. The Attu requires this privilege to show the database list.

Then login with dbbuser, you will see an error "PrivilegeSelectOwnership", but it doesn't matter. You can switch to dbb database and do the operations with the privileges that you have.
